src/os/ec_bsd.c

```diff
@@ -136,7 +136,7 @@ u_int16 get_iface_mtu(const char *iface)
 
    /* open the socket to work on */
    sock = socket(PF_INET, SOCK_DGRAM, 0);
-   if (sock = -1)
+   if (sock == -1)
```

Comment:
I don't know how the FreeBSD guys build  the package. I learned, that they take the existing source and patch from upstream. However this would normally also end in build issues.
However I started to test ettercap on FreeBSD and faced all the different addressed issues.

This specific issue was just a warning. However the actual functionality was obviously not given.